### PR TITLE
Updated interfaces to easy creating

### PR DESCRIPTION
**Pull request by AnttiKauppila**
- Code is organized a bit better way
 * Connectivity logging is moved to own function
 * Connectivity setup is moved to own function

- Readme updated according to changes done

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,34 +45,33 @@ With Mbed OS 5.9, the EMAC SW was refactored and a default network selector is u
...
-            "value": "WIFI_ESP8266"
-        },
-        "wifi-ssid": {
-            "value": "\"SSID\""
-        },
-        "wifi-password": {
-            "value": "\"Password\""
+    "target_overrides": {
+        "*": {
+            "easy-connect.wifi-type": "\"ESP8266\"",
```
Comment (JanneKiiskila) [inline README.md]: Seems like duplication that you have to tell it so many times, it's ESP8266? Doesn't seem right to me. One selection MUST be enough and then rest of it needs to happen outside of the `mbed_app.json`.

File: mcr20a-rf-driver.lib [hunk 2]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/mcr20a-rf-driver/#93661a696735279590378e6abef1d689799b713e
+https://github.com/ARMmbed/mcr20a-rf-driver/#17c467b64ab604887fc160cbe4118a51477e334b
```
Comment (JanneKiiskila) [inline mcr20a-rf-driver.lib:1]: The .lib changes should have been in another PR, IMHO.

File: mbed_lib.json [hunk 3]
```diff
@@ -1,62 +1,62 @@
...
-            "help": "Enable debug logs for (ESP8266)",
-            "value": false
+        "network-interface-class": {
+            "help": "The class name for interface",
+            "value": null
         },
-        "wifi-idw01m1-tx": {
-            "help": "TX pin for serial connection to external device (X-NUCLEO-IDW01M1)",
-            "value": "PA_9"
+        "network-interface-name": {
```
Comment (JanneKiiskila) [inline mbed_lib.json]: This is too much repetition,  IMHO.

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]

**Comment by AnttiKauppila** [comment]
@JanneKiiskila Cleaned up the configuration as you suggested, please re-review

**Pull request closed without merging**